tapgarden/caretaker.go

```diff
@@ -1148,6 +1148,17 @@ func (b *BatchCaretaker) stateStep(currentState BatchState) (BatchState, error)
 			return 0, fmt.Errorf("unable to confirm batch: %w", err)
 		}
 
+		// Send supply commitment events for all minted assets before
+		// finalizing the batch. This ensures that supply commitments
+		// are tracked before the batch is considered complete.
+		err = b.sendSupplyCommitEvents(
```

Comment:
Should we do this before we call `MarkBatchConfirmed`? If we shut down while sending the supply commit events, the batch will already be marked as confirmed in the DB, so it won't be re-tried on next startup (and things aren't done atomically here).

Reply:
Makes sense, swapped.